addons/website_event/models/event_event.py

```diff
@@ -387,15 +379,25 @@ def _update_website_menu_entry(self, fname_bool, fname_o2m, fmenu_type):
                      if menu_info[4] == fmenu_type]
         if self[fname_bool] and not self[fname_o2m]:
             # menus not found but boolean True: get menus to create
-            for name, url, xml_id, menu_sequence, menu_type in menu_data:
-                new_menu = self._create_menu(menu_sequence, name, url, xml_id, menu_type)
+            for name, url, xml_id, menu_sequence, menu_type, parent_menu_type in menu_data:
+                new_menu = self._create_menu(menu_sequence, name, url, xml_id, menu_type, parent_menu_type)
         elif not self[fname_bool]:
-            # will cascade delete to the website.event.menu
-            self[fname_o2m].mapped('menu_id').sudo().unlink()
+            # Get the menus that are being unlinked
+            menus_to_unlink = self[fname_o2m].mapped('menu_id')
+
+            # Iterate through each menu being unlinked
+            for menu in menus_to_unlink:
+                # Check if the menu has any child menus
+                child_menus = self.env['website.menu'].search([('parent_id', '=', menu.id)])
+                if child_menus:
+                    child_menus.write({'parent_id': menu.parent_id.id})
```

Comment:
here is the scenario:
Install Booth first
-> Exhibitors Parent Menu will created with 'booth' menu_type
Install Exhibitors 2nd
-> Exhibitors will be place inside 'Exhibitors Parent Menu' created by Booth

Now if the user wants to remove Booth, Both 'Exhibitors Parent Menu' and 'Booth Submenu' will disappear.
The question is what will happen to 'Exhibitors' submenu, It will also disappear. 
Hence we remove parent menu_id (which is Exhibitors Parent Menu) from it and set event menu as menu_id, so that it work like a independent menu.

when you delete a directory on your PC, before deleting you moved out a single file from that directory.